### ISSUE TITLE
--exclude-unless-present

### ISSUE THREAD

**Issue by denizdogan** [rclone/rclone#4753]
#### What is your current rclone version (output from `rclone version`)?

rclone v1.53.2-DEV
- os/arch: darwin/amd64
- go version: go1.15.3

#### What problem are you are trying to solve?

We have a system where we need to copy files from an SFTP server to a Google Cloud Storage bucket. Our customers upload files into subdirectories on the SFTP server and when they're done, they upload an empty file named `foobar_{timestmap}.xml`. We wish to transfer the entire sub-directory from the SFTP server to GCS only once this file has been uploaded.

#### How do you think rclone should be changed to solve that?

I was thinking `--exclude-unless-present` (in line with `--exclude-if-present`) or possibly `--include-if-present` (although I think the former naming is more accurate).

**Comment by ivandeex** [comment]
Such logic is traditionally implemented in ad-hoc workflow scripts. This feature doesn't belong with rclone by nature, I believe.

**Comment by denizdogan** [comment]
@ivandeex Couldn't you argue the same about already existing features such as `--exclude-if-present`?

**Comment by ivandeex** [comment]
I can't find an issue that would explain reasons why the flag was added. IMHO, if it helped to fix a storage provider issue then it's reasonable. Otherwise, not.
We'd better wait for @ncw here.

Currently rclone can be seen as a swiss army knife for opetating cloud storages. If somebody decides to reposition it as a generic file workflow tool, then ok, we'd better start laying out internal api and config formats to express chains of events, conditions and actions.

**Comment by ivandeex** [comment]
This particular use case does not seem to be related to a problem with a storage backend, nor is it generic enough.

**Comment by ivandeex** [comment]
Request captured in https://github.com/rclone/rclone/issues/4810#issuecomment-974184110

Superseded by #4810
